commit ad4959a43a6b61cf48f2f3bf92879e1077decd9e
Author: Matthias Reichl <hias@horus.com>
Date:   Fri Feb 2 12:00:00 2018 +0100

    Revert downstream wm8804 changes
    
    The format change from S24_LE to S24_3LE effectively disables 24-bit
    mode as S24_3LE isn't supported by bcm2835-i2s. This causes issues
    with drivers that want to use wm8804 in 24-bit mode.
    
    Adding the S32_LE format is also incorrect, according to the datasheet
    only 16-24 bit formats are supported.
    
    Signed-off-by: Matthias Reichl <hias@horus.com>

diff --git a/sound/soc/codecs/wm8804.c b/sound/soc/codecs/wm8804.c
index d8fdce81b297..fc69b87443d8 100644
--- a/sound/soc/codecs/wm8804.c
+++ b/sound/soc/codecs/wm8804.c
@@ -304,7 +304,6 @@ static int wm8804_hw_params(struct snd_pcm_substream *substream,
 		blen = 0x1;
 		break;
 	case 24:
-	case 32:
 		blen = 0x2;
 		break;
 	default:
@@ -516,7 +515,7 @@ static const struct snd_soc_dai_ops wm8804_dai_ops = {
 };
 
 #define WM8804_FORMATS (SNDRV_PCM_FMTBIT_S16_LE | SNDRV_PCM_FMTBIT_S20_3LE | \
-			SNDRV_PCM_FMTBIT_S24_3LE | SNDRV_PCM_FMTBIT_S32_LE)
+			SNDRV_PCM_FMTBIT_S24_LE)
 
 #define WM8804_RATES (SNDRV_PCM_RATE_32000 | SNDRV_PCM_RATE_44100 | \
 		      SNDRV_PCM_RATE_48000 | SNDRV_PCM_RATE_64000 | \
@@ -544,7 +543,7 @@ static struct snd_soc_dai_driver wm8804_dai = {
 };
 
 static const struct snd_soc_codec_driver soc_codec_dev_wm8804 = {
-	.idle_bias_off = false,
+	.idle_bias_off = true,
 
 	.component_driver = {
 		.dapm_widgets		= wm8804_dapm_widgets,
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1140_ad4959a43a6b61cf48f2f3bf92879e1077decd9e.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1140_ad4959a43a6b61cf48f2f3bf92879e1077decd9e.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
